syft/tensor.py

```diff
@@ -2994,6 +2994,33 @@ def mv(self, tensorvector):
             raise NotImplemented
         return mv(self, tensorvector)
 
+    def narrow(self, dim, start, length):
+        """
+        Returns a new tensor that is a narrowed version of this tensor.
+        The dimension ``dim`` is narrowed from ``start`` to ``start`` + ``length``.
+
+        :param dim: dimension along which to narrow
+        :param start: starting dimension
+        :param length: length from start to narrow to
+        :return: narrowed version of this tensor
+        """
+        dim = dim if dim >= 0 else dim + self.dim()
+        if self.encrypted:
+            raise NotImplemented
+        if not isinstance(dim, int):
+            raise TypeError("The value of dimension must be integer")
+        if not isinstance(start, int):
+            raise TypeError("The value of start must be integer")
+        if not isinstance(length, int):
+            raise TypeError("The value of length must be integer")
```

Comment:
why not something like this: 

    if not isinstance(dim, int) or not isinstance(start, int) or not isinstance(length, int):
        raise TypeError(f"narrow received an invalid combination of arguments - got {dim.__class__.__name__},
        {start.__class__.__name__},{length.__class__.__name__},but expected (int dim, int start, int length")

so that if you pass multiple invalid arguments you can point them out clearly, instead of the current code, where you cant do that.

Reply:
Agree, that would be much better; I've updated the PR.